app/dao/provider_details_dao.py

```diff
@@ -34,57 +33,92 @@ def dao_get_provider_versions(provider_id):
     ).all()
 
 
-@transactional
-def dao_reduce_sms_provider_priority(identifier, *, time_threshold):
-    """
-    Will reduce a chosen sms provider's priority, and increase the other provider's priority by 10 points each.
-    If either provider has been updated in the last `time_threshold`, then it won't take any action.
+def _adjust_provider_priority(provider, new_priority):
+    print('Adjusting provider priority - {} going from {} to {}'.format(
+        provider.identifier,
+        provider.priority,
+        new_priority,
+    ))
+    provider.priority = new_priority
+
+    # Automatic update so set as notify user
+    provider.created_by_id = current_app.config['NOTIFY_USER_ID']
+
+    # update without commit so that both rows can be changed without ending the transaction
+    # and releasing the for_update lock
+    _update_provider_details_without_commit(provider)
+
+
+def _get_sms_providers_for_update(time_threshold):
     """
-    amount_to_reduce_by = 10
+    Returns a list of providers, while holding a for_update lock on the provider details table, guaranteeing that those
+    providers won't change (but can still be read) until you've committed/rolled back your current transaction.
 
+    if any of the providers have been changed recently, it returns an empty list - it's still your responsiblity to
+    release the transaction in that case
+    """
     # get current priority of both providers
     q = ProviderDetails.query.filter(
         ProviderDetails.notification_type == 'sms',
         ProviderDetails.active
     ).with_for_update().all()
 
-    providers = {provider.identifier: provider for provider in q}
-    other_identifier = get_alternative_sms_provider(identifier)
-
     # if something updated recently, don't update again. If the updated_at is null, treat it as min time
     if any((provider.updated_at or datetime.min) > datetime.utcnow() - time_threshold for provider in q):
```

Comment:
Optional: I feel that `_get_sms_providers_for_update` is doing too much. I think it could just return the provider it has found and then a different method is responsible for returning a boolean of if providers have been updated too recently.